dove/tests/test_cmd_dove_ct.rs

```diff
@@ -0,0 +1,320 @@
+use fs_extra::file::write_all;
+use dove::cmd::ct::Transaction;
+
+mod helper;
+use crate::helper::{project_start_new_and_build, project_remove, execute_dove_at};
+
+/// $ dove ct
+#[test]
+fn test_cmd_dove_ct_without_arguments() {
+    // Path to dove folder, project and project name
+    let project_name = "demoproject_19";
+    let project_folder = project_start_new_and_build(project_name);
+
+    // project_folder/scripts/sdemo.move
+    write_all(
+        &project_folder.join("scripts/sdemo.move"),
+        "script {
+                    fun main() {
+                        assert((1+3)==4,1);
+                    }
+                }",
+    )
+    .unwrap();
+
+    let args = &["dove", "ct"];
+    execute_dove_at(&project_folder, args).unwrap_or_else(|err| {
+        panic!("{}", err);
+    });
+    let tx_path = project_folder.join("target/transactions/main.mvt");
+    assert!(
+        tx_path.exists(),
+        "Transaction not found: {}\r\n[Command] {}",
+        tx_path.display(),
+        args.join(" "),
+    );
+    let tx_fmt = format!(
+        "{:?}",
+        bcs::from_bytes::<Transaction>(std::fs::read(&tx_path).unwrap().as_ref()).unwrap()
+    );
+    assert!(tx_fmt.contains(" args: []"));
+    assert!(tx_fmt.contains(" type_args: []"));
+    assert!(tx_fmt.contains(" signers_count: 0"));
+
+    project_remove(&project_folder);
+}
+/// $ dove ct 'sdemo_4<u8>(16)'
+#[test]
+fn test_cmd_dove_ct_with_type() {
+    // Path to dove folder, project and project name
+    let project_name = "demoproject_24";
+    let project_folder = project_start_new_and_build(project_name);
+
+    // project_folder/modules/mdemo.move
+    write_all(
+        &project_folder.join("modules/mdemo.move"),
+        "module ModuleDemo{
+            struct T1 {}
+            struct T2 {}
+
+            struct Demo<T> has drop{
+                value:u8
+            }
+            public fun new<T:drop>(value:u8): Demo<T>{
+                Demo<T>{
+                    value
+                }
+            }
+        }",
+    )
+    .unwrap();
+    // project_folder/scripts/sdemo.move
+    write_all(
+        &project_folder.join("scripts/sdemo.move"),
+        "script {
+            use 0x1::ModuleDemo;
+
+            fun sdemo_4<T:drop>(value:u8) {
+                let _tmp:ModuleDemo::Demo<T> = ModuleDemo::new<T>(value);
+            }
+        }",
+    )
+    .unwrap();
+    let args = &["dove", "ct", "sdemo_4<u8>(16)"];
+    execute_dove_at(&project_folder, args).unwrap_or_else(|err| {
```

Comment:
This `unwrap_or_else` call duplicated everywhere. I think plain `unwrap()` should suffice as it will print error on fail anyway.

Reply:
The output is different.

![image](https://user-images.githubusercontent.com/32911557/119544871-9910bb80-bdab-11eb-8092-fbfdeb0ff4f2.png)
![image](https://user-images.githubusercontent.com/32911557/119544938-a9c13180-bdab-11eb-8f6c-32e006d4a5f4.png)

Reply:
Then I would suggest removing these lines
https://github.com/pontem-network/move-tools/pull/34/files#diff-f25bd94ddb1dd30d834b28410ca2d8840a21c59afc2b0e487a59e7becfbf3d9eR128-R142
and instead, create a function like
```
fn panic_with_formatted_error(err)
```
which will have formatting from those lines added, and replace all current `unwrap_or_else` calls arguments with this function. 

Trying to understand why you wrap error string into those `[FOLDER] / [ERROR]` formatting so early. Don't you need a raw error string on the upper level to check it first, and then apply formatting?